src/util/lang_object.js

```diff
@@ -63,7 +63,7 @@
 
   //TODO: this function return an empty object if you try to clone null
   function clone(object, deep) {
-    return extend({ }, object, deep);
+    return deep ? extend({ }, object, deep) : Object.assign({}, object);
```

Comment:
since we removed support for ie11 and we plan to add a transpiler, i think we can leverage Object.assign for the shallow clone.